### DISCUSSION TITLE
v5.46.0

### DISCUSSION THREAD

**Discussion by sokra** [webpack/webpack#13839] [General]
# Features

* status handlers in HMR api can now return Promises to delay the HMR process
* reasons in stats can now be grouped and collapsed
  * add `stats.reasonsSpace` and `stats.groupReasonsByOrigin`

# Bugfixes

* fix a crash in asset modules when updating persistent cached modules from unsafe cached modules

# Performance

* detailed preset limits all spaces to 1000 by default
* upgrade webpack-sources for a performance bugfix

<hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.46.0'>v5.46.0</a>.</em>

**Comment by tiye** [comment]
Got problem after upgraded from `5.45.1`. I'm using Webpack HMR for Node.js with configs https://github.com/calcit-lang/editor/blob/main/webpack.config.js . After upgraded, the process keeps logging. Any clue why it's happening. Sorry I can't locate where's the problem, only narrowed down to this upgrade.

```
=>> yarn outdated
yarn outdated v1.22.4
Package        Current Wanted Latest Package Type    URL
latest-version 5.1.0   5.1.0  6.0.0  dependencies    https://github.com/sindresorhus/latest-version#readme
webpack        5.45.1  5.46.0 5.46.0 devDependencies https://github.com/webpack/webpack
✨  Done in 3.51s.
=>> yarn upgrade webpack
yarn upgrade v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Rebuilding all packages...
success Saved lockfile.
success Saved 54 new dependencies.
info Direct dependencies
└─ webpack@5.46.0
info All dependencies
├─ @types/eslint-scope@3.7.1
├─ @types/eslint@7.28.0
├─ @types/json-schema@7.0.8

......a lot more

├─ uri-js@4.4.1
├─ watchpack@2.2.0
├─ webpack-sources@2.3.1
├─ webpack@5.46.0
└─ yocto-queue@0.1.0
✨  Done in 11.10s.
=>> node ../js-out/bundle.js
[Warning] versions mismatch, CLI using: 0.4.10, runtime using: 0.4.11
(get-env "mode"): undefined
(get-env "memofVerbose"): undefined
Loading calcit.cirru
Took 6ms to load.
port 6001 is in use.
port 6002 is ok, please edit on http://calcit-editor.cirru.org?port=6002
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.
[HMR] Nothing hot updated.

endlessly...
```

my js code was generated from my own language which like CoffeeScript and ClojureScript. didn't retry with a minimal project.

**Comment by tiye** [comment]
BTW do you have other recommendations for HMR on Node.js after so many years? I feel it's not quite correct to install a whole Webpack and only use it for Node.js HMR, in such an age that people talking about ES6 npm modules.

**Reply by tiye** [reply]
I'm kind of feeling that users like me who are using tricky and legacy features of Webpack are blocking the process of modernizing Webpack. 😅
